sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,544 @@
+    item_price: FieldValue
+    total_price: FieldValue
+
+class FieldValue:
+    value: Union[str, float, int, datetime]
+    text: str
+    bounding_box: List[float]
+    confidence: float
+    page_number: int
+    elements: List[Union[ExtractedLine, ExtractedWord]]
+
+class ExtractedLine:
+    text: str
+    bounding_box: List[float]
+    language: str
+    page_number: int
+    words: List[ExtractedWord]
+
+class ExtractedWord:
+    text: str
+    bounding_box: List[float]
+    confidence: float
+    page_number: int
+    line: ExtractedLine
+
+class PageMetadata:
+    page_number: int
+    angle: float
+    width: int
+    height: int
+    unit: str
+    language: str
+    lines: List[ExtractedLine]
+
+PageRange = collections.namedtuple("PageRange", "first_page last_page")
+```
+
+### Receipt Sample
+```python
+from azure.ai.formrecognizer import FormRecognizerClient
+
+client = FormRecognizerClient(endpoint, credential)
+
+receipt_image = "https://raw.githubusercontent.com/Azure-Samples/cognitive-services-REST-api-samples/master/curl/form-recognizer/contoso-allinone.jpg"
+poller = client.begin_extract_receipt(receipt_image)
+receipt = poller.result()
+r = receipt[0]
+
+print("Receipt contained the following values with confidences: ")
+print("ReceiptType: {}, confidence: {}").format(r.receipt_type.value, r.receipt_type.confidence)
+print("MerchantName: {}, confidence: {}").format(r.merchant_name.value, r.merchant_name.confidence)
+print("MerchantAddress: {}, confidence: {}").format(r.merchant_address.value, r.merchant_address.confidence)
+print("MerchantPhoneNumber: {}, confidence: {}").format(r.merchant_phone_number.value, r.merchant_phone_number.confidence)
+print("TransactionDate: {}, confidence: {}").format(r.transaction_date.value, r.transaction_date.confidence)
+print("TransactionTime: {}, confidence: {}").format(r.transaction_time.value, r.transaction_time.confidence)
+for item in r.receipt_items:
+    print("Item Name: {}, confidence: {}").format(item.name.value, item.name.confidence)
+    print("Item Quantity: {}, confidence: {}").format(item.quantity.value, item.quantity.confidence)
+    print("Item Price: {}, confidence: {}").format(item.item_price.value, item.item_price.confidence)
+    print("Total Price: {}, confidence: {}").format(item.total_price.value, item.total_price.confidence)
+print("Subtotal: {}, confidence: {}").format(r.subtotal.value, r.subtotal.confidence)
+print("Tax: {}, confidence: {}").format(r.tax.value, r.tax.confidence)
+print("Tip: {}, confidence: {}").format(r.tip.value, r.tip.confidence)
+print("Total: {}, confidence: {}").format(r.total.value, r.total.confidence)
+
+# Access as a dictionary
+for item, field_value in r.items():
+    print(item, field_value.value, field_value.text, field_value.confidence, field_value.bounding_box)
+```
+
+### Layout Models
+
+```python
+class ExtractedLayoutPage:
+    tables: List[ExtractedTable]
+    page_metadata: PageMetadata
+    page_number: int
```

Comment:
Is `page_number` redundant with `page_metadata.page_number`?

Reply:
Yes, I suppose. But since this class represents a page, I think it makes sense to have this as a top-level attribute.